epitator/structured_incident_annotator.py

```diff
@@ -0,0 +1,287 @@
+            tables.append(Table(
+                column_definitions,
+                rows,
+                metadata=dict(
+                    __section_title=__section_title,
+                    __last_geoname_mentioned=__last_geoname_mentioned,
+                    __last_date_mentioned=__last_date_mentioned)))
+
+        incidents = []
+        for table in tables:
+            for row_idx, row in enumerate(table.rows):
+                row_incident_date = table.metadata.get('__last_date_mentioned')
+                row_incident_location = table.metadata.get('__last_geoname_mentioned')
+                row_incident_base_type = None
+                row_incident_status = None
+                row_incident_aggregation = None
+                for column, value in zip(table.column_definitions, row):
+                    if not value: continue
+                    if column['type'] == 'date':
+                        row_incident_date = value
+                    elif column['type'] == 'geoname':
+                        row_incident_location = value
+                    elif column['type'] == 'incident_type':
+                        if "case" in value.text.lower():
+                            row_incident_base_type = "caseCount"
+                        elif "death" in value.text.lower():
+                            row_incident_base_type = "deathCount"
+                    elif column['type'] == 'incident_status':
+                        row_incident_status = value.text
+
+                row_incidents = []
+                for column, value in zip(table.column_definitions, row):
+                    if not value: continue
+                    if column['type'] == "number":
+                        column_name = column.get('name', '').lower()
+                        if row_incident_base_type:
+                            incident_base_type = row_incident_base_type
+                        else:
+                            if "cases" in column_name:
+                                incident_base_type = "caseCount"
+                            elif "deaths" in column_name:
+                                incident_base_type = "deathCount"
+                            else:
+                                incident_base_type = "caseCount"
+                        if row_incident_status:
+                            count_status = row_incident_status
+                        else:
+                            if "suspect" in column_name:
+                                count_status = "suspected"
+                            elif "confirmed" in column_name:
+                                count_status = "confirmed"
+                            else:
+                                count_status = None
+                        incident_aggregation = None
+                        if row_incident_aggregation is not None:
+                            incident_aggregation = row_incident_aggregation
```

Comment:
I'm guessing this is to allow a specific type of aggregation to be passed in?

Reply:
Yes, it will be used for tables that are labeled as being cumulative cases.